lib/handlers/addProduct.ts

```diff
@@ -0,0 +1,18 @@
+import { ProductService } from '../services/productService';
+import { ProductRepository } from '../repositories/productRepository';
+import { Product } from '../models/product';
+
+export async function main(event: any): Promise<Product> {
+    const repo = new ProductRepository();
+    const service = new ProductService(repo);
```

Comment:
Handler's concern should just be calling a service. We should never have to reference a repository here. Especially since there's no need to inject it here, we could just initialize it in the service.

Reply:
Taken care for all handlers of product, category, order